### PR TITLE
sdcv: new package

### PR DESCRIPTION
**Pull request by deeruarty**
- updated test scripts for XDG under mingw
- stat -> stat64/stat32 for mingw ifdefs
- c++11 -> gnu++20
-  cleanup test dicts after tests run

### CODE REVIEW

File: mingw-w64-sdcv/001-sdcv-0.5.3-fix-mingw-build.patch [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+     const std::list<std::string> vars = get_cache_variant(url);
+ 
+     for (const std::string &item : vars) {
++
++#ifdef __MINGW64__
++        struct ::_stat64 idxstat, cachestat;
++#elif __MINGW32__
++        struct ::_stat32 idxstat, cachestat;
++#elif __MSYS__
+         struct ::stat idxstat, cachestat;
```
Comment (lazka) [inline mingw-w64-sdcv/001-sdcv-0.5.3-fix-mingw-build.patch:81]: This should really use `GStatBuf` in all cases.

File: mingw-w64-sdcv/001-sdcv-0.5.3-fix-mingw-build.patch [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+index 15da246..dced712 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -119,7 +119,8 @@ if (ENABLE_NLS)
+   install(DIRECTORY "${gettext_outDir}" DESTINATION "share")
+ endif ()
+ 
+-option(BUILD_TESTS "Enable automatic testing" False)
++#option(BUILD_TESTS "Enable automatic testing" False)
++option(BUILD_TESTS "Enable automatic testing" True)
```
Comment (lazka) [inline mingw-w64-sdcv/001-sdcv-0.5.3-fix-mingw-build.patch:11]: What's this about?

File: mingw-w64-sdcv/001-sdcv-0.5.3-fix-mingw-build.patch [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+ 
++    /* fprintf(stdout, "UserHomeDir: %s\n", std::string(g_get_home_dir()).c_str()); */
++    /* fprintf(stdout, "UserDataDir: %s\n", std::string(g_get_user_data_dir()).c_str()); */
++
+     std::string conf_dir = std::string(g_get_home_dir()) + G_DIR_SEPARATOR + ".stardict";
+-    if (!g_file_test(conf_dir.c_str(), G_FILE_TEST_IS_DIR))
+-        conf_dir = std::string(g_get_user_data_dir()) + G_DIR_SEPARATOR + "stardict";
++    if (!g_file_test(conf_dir.c_str(), G_FILE_TEST_IS_DIR)) {
++
++#ifdef __MSYS2__
```
Comment (lazka) [inline mingw-w64-sdcv/001-sdcv-0.5.3-fix-mingw-build.patch:44]: What's `__MSYS2__`?

File: mingw-w64-sdcv/001-sdcv-0.5.3-fix-mingw-build.patch (hunk 2 above)
Comment (deeruarty) [reply mingw-w64-sdcv/001-sdcv-0.5.3-fix-mingw-build.patch:11]: build unittests by default without additional ```cmake param -DBUILD_TESTS```.

File: mingw-w64-sdcv/001-sdcv-0.5.3-fix-mingw-build.patch (hunk 3 above)
Comment (deeruarty) [reply mingw-w64-sdcv/001-sdcv-0.5.3-fix-mingw-build.patch]: mistype, fixed.

File: mingw-w64-sdcv/001-sdcv-0.5.3-fix-mingw-build.patch (hunk 1 above)
Comment (deeruarty) [reply mingw-w64-sdcv/001-sdcv-0.5.3-fix-mingw-build.patch]: fixed

File: mingw-w64-sdcv/001-sdcv-0.5.3-fix-mingw-build.patch (hunk 2 above)
Comment (lazka) [reply mingw-w64-sdcv/001-sdcv-0.5.3-fix-mingw-build.patch:11]: But why is this change needed, if you can just set it via "-D"?

Comment (deeruarty) [reply mingw-w64-sdcv/001-sdcv-0.5.3-fix-mingw-build.patch]: ok, fixed to -DBUILD_TESTS https://github.com/msys2/MINGW-packages/commit/0c84c629595c7e3628367131dbab4439ba4e2081

File: mingw-w64-sdcv/PKGBUILD [hunk 4]
```diff
@@ -1,17 +1,22 @@
...
-mingw_arch=('mingw32' 'mingw64')
+mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32')
 url="https://dushistov.github.io/sdcv/"
 license=('GPL')
-depends=("${MINGW_PACKAGE_PREFIX}-glib2" "libreadline" "${MINGW_PACKAGE_PREFIX}-zlib")
-makedepends=("${MINGW_PACKAGE_PREFIX}-cmake" "${MINGW_PACKAGE_PREFIX}-ninja" "${MINGW_PACKAGE_PREFIX}-glib2")
-provides=('stardict')
+depends=(
+  "${MINGW_PACKAGE_PREFIX}-glib2"
+  "${MINGW_PACKAGE_PREFIX}-libreadline"
```
Comment (deeruarty) [inline mingw-w64-sdcv/PKGBUILD:15]: it look there is no libreadline for mingw...
Is there any alternatives? Or sdcv should be patched to remove this dependency?

Comment (lazka) [reply mingw-w64-sdcv/PKGBUILD:15]: oh, typo, there is a "readline" package containing everything

Comment (lazka) [reply mingw-w64-sdcv/PKGBUILD]: (mingw packages can't use msys/cygwin libraries, at runtime at least)

Comment (deeruarty) [reply mingw-w64-sdcv/PKGBUILD]: oh, just mingw64-readline, ok.

**Pull request merged**